### ISSUE TITLE
lockfile is tied to python version

### ISSUE THREAD

**Issue by DetachHead** [pyprojectx/pyprojectx#85]
**Describe the bug**
my project is supposed to work on python versions 3.8 to 3.12, however it doesn't seem to be possible to generate a lockfile that works on multiple python versions:

```
ERROR: Ignored the following versions that require a different python version: 0.1.0 Requires-Python >= 3.10; 0.2.0 Requires-Python >= 3.10; 0.3.0 Requires-Python >= 3.10; 0.4.0 Requires-Python >= 3.10; 0.5.0 Requires-Python >= 3.10; 0.6.0 Requires-Python >= 3.10; 0.6.1 Requires-Python >= 3.10; 0.7.0 Requires-Python >= 3.10; 0.8.0 Requires-Python >= 3.10
ERROR: Could not find a version that satisfies the requirement truststore==0.8.0 (from versions: none)
ERROR: No matching distribution found for truststore==0.8.0

Notice:  A new release of pip is available: 23.3.1 -> 24.0
Notice:  To update, run: python -m pip install --upgrade pip
PYPROJECTX ERROR: installation of 'pdm' failed with exit code 1
```

**To Reproduce**
1. put `pdm` in your pyprojectx dependencies:
    ```toml
    # pyproject.toml
    [tool.pyprojectx]
    pdm = "pdm>=2.12.3"
    ```
2. run `./pw --lock` using python 3.12
3. run `./pw pdm --version` using python 3.8

**Expected behavior**
no error: generated lockfile should contain dependency information for all supported python versions

**Desktop (please complete the following information):**
 - Python  version: 3.8, 3.12
 - OS: windows, linux, macos

**Additional context**
full output: https://github.com/DetachHead/pytest-robotframework/actions/runs/7999782115/job/21848170675?pr=217

**Comment by houbie** [comment]
I added a clarification in the docs:

> Supporting multiple python versions:
> Make sure to run px --lock using the minimum supported python version for your project. This prevents the lock file from containing dependencies that are incompatible with older python versions.

Although not perfect, locking does avoid [broken toolchains](https://upcycled-code.com/blog/the-broken-version-breakdown/).

This could be improved by using a better dependency locking mechanism than just plain `pip freeze`.
I considered using Pdm's or Poetry's dependency locking libs, but that's not planned in the near future.

**Comment by KotlinIsland** [comment]
Is this resolved in 3.0?

**Comment by houbie** [comment]
Indeed resolved in 3.0 by using uv
